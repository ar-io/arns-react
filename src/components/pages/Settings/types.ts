export interface NetworkSettingsState {
  gateway: string;
  cuUrl: string;
  muUrl: string;
  suAddress: string;
  registryAddress: string;
  antRegistryAddress: string;
  hyperbeamUrl: string;
  turboPaymentUrl: string;
  showGatewayModal: boolean;
}

export interface NetworkSettingsValidation {
  gateway: boolean;
  cuUrl: boolean;
  muUrl: boolean;
  suAddress: boolean;
  registryAddress: boolean;
  antRegistryAddress: boolean;
  hyperbeamUrl: boolean;
  turboPaymentUrl: boolean;
}

export type NetworkSettingsAction =
  | {
      type: 'SET_VALUE';
      field: keyof NetworkSettingsState;
      value: string | boolean;
    }
  | {
      type: 'SET_VALIDATION';
      field: keyof NetworkSettingsValidation;
      isValid: boolean;
    }
  | { type: 'RESET_TO_DEFAULTS' }
  | { type: 'SET_TESTNET_DEFAULTS' }
  | { type: 'SET_MAINNET_DEFAULTS' }
  | { type: 'SYNC_FROM_GLOBAL_STATE'; payload: Partial<NetworkSettingsState> };

export interface SettingConfig {
  key: keyof NetworkSettingsState;
  label: string;
  placeholder: string;
  validator: (value: string) => boolean;
  defaultValue: string;
  testnetValue?: string;
  mainnetValue?: string;
  type: 'url' | 'transaction-id' | 'gateway';
  onSet?: (value: string) => void;
  onReset?: () => void;
  globalStateSync?: {
    getValue: () => string;
    setValue: (value: string) => void;
  };
}
