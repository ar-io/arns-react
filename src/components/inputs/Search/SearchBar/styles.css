.searchbar-container {
  display: flex;
  flex-direction: column;
  width: 100%;
  max-width: 1200px;
  justify-content: center;
  align-items: center;
  gap: 3em;
}

.searchbar {
  display: flex;
  position: relative;
  width: 100%;
  max-width: 920px;
  min-width: 250px;
  background-color: var(--card-bg);
  border: 2px solid transparent;
  border-radius: var(--corner-radius);
  align-items: center;
  justify-content: space-between;
  box-sizing: border-box;
}

.searchbar input {
  box-sizing: border-box;
  display: flex;
  width: 100%;
  font-size: 16px;
  font-weight: 400;
  border: none;
  outline: none;
  padding-left: 26px;
  padding-right: 26px;
  background-color: var(--card-bg);
  color: var(--text-white);
}
.searchbar input::placeholder {
  font-family: 'Rubik';
  font-style: normal;
  font-weight: 400;
  font-size: 16px;
  color: var(--placeholder-text);
  text-transform: none;
}

input[type='search']::-webkit-search-decoration {
  display: none;
}
input[type='search']::-webkit-search-cancel-button {
  display: none;
}

.searchbar input:-webkit-autofill,
.searchbar input:-webkit-autofill:hover,
.searchbar input:-webkit-autofill:focus,
.searchbar input:-webkit-autofill:active {
  -webkit-text-fill-color: white;
  -webkit-box-shadow: 0 0 0px 1000px var(--card-bg) inset;
}

@media screen and (max-width: 900px) {
  .searchbar {
    max-width: 600px;
  }
}

@media screen and (max-width: 600px) {
  .searchbar input {
    text-align: center;
  }
  .searchbar {
    max-width: 300px;
  }
}
