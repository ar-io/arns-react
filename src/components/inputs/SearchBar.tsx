import searchIcon from '../../images/DarkMode/SearchIcon.png'


type searchBar = {
    searchButtonAction: any
}

function SearchBar({searchButtonAction}: searchBar){

    return (

        <div className="searchBar">
        <input type="text" placeholder='click search button to toggle the search state' />
        <button className='searchButton' onClick={()=> {searchButtonAction()}}><img src={searchIcon} /></button>
        </div>
    )
}

export default SearchBar