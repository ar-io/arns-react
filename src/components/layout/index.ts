import FeaturedDomains from './FeaturedDomains/FeaturedDomains';
import Layout from './Layout/Layout';
import Loader from './Loader/Loader';
import NavBarLink from './Navbar/NavBarLink/NavBarLink';
import NavGroup from './Navbar/NavGroup/NavGroup';
import NavBar from './Navbar/Navbar';
import ProtectedRoute from './ProtectedRoute/ProtectedRoute';

export {
  FeaturedDomains,
  Layout,
  Loader,
  NavBar,
  NavBarLink,
  NavGroup,
  ProtectedRoute,
};
