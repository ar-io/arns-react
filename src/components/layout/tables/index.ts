import AntTable from './AntTable/AntTable';

export { AntTable };
