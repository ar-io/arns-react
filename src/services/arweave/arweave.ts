import { ArweaveTransactionID } from '../../types';

export const buildContractTxQuery = (id: ArweaveTransactionID) => {
  const queryObject = {
    query: `{
    transactions(first: 100, sort: HEIGHT_DESC,
      tags: [],
    ids:["${id.toString()}"]) {
      edges {
        node {
          id
          owner {
            address
          }
          data {
            size
          }
          block {
            height
            timestamp
          }
          tags {
            name,
            value
          }
        }
      }
    }
  }`,
  };
  return queryObject;
};
