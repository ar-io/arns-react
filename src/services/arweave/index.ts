import Arweave from 'arweave/node/common';

import { SmartweaveContractSource } from '../../types';
import { ArweaveCompositeDataProvider } from './ArweaveCompositeDataProvider';
import { WarpDataProvider } from './WarpDataProvider';

export const defaultDataProvider = (
  arweave: Arweave,
): SmartweaveContractSource =>
  new ArweaveCompositeDataProvider([new WarpDataProvider(arweave)], arweave);
