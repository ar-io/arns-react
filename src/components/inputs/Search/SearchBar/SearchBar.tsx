import { ReactComponent as SearchIcon } from '../../../icons/Search.svg';
import './styles.css';
import { SearchBarProps } from '../../../../types';
import React, { useEffect, useState } from 'react';

function SearchBar(props: SearchBarProps) {
  
  const {
    buttonAction,
    placeholderText,
    headerElement,
    footerText
  } = props;

  
  const [searchBarText, setSearchBarText] = useState('');
  const [isValid, setIsValid] = useState<Boolean | undefined>(undefined);

 

  function onHandleChange(name:string) {
    setSearchBarText(name);
  }

  return (
    <>
      {headerElement}
      <div
        className="searchBar"
        style={
          isValid
            ? { borderColor: 'var(--success-green)' }
            : isValid === false
            ? { borderColor: 'var(--error-red)' }
            : isValid === undefined
            ? { borderColor: '' }
            : {}
        }
      >
        <input
          type="text"
          placeholder={placeholderText}
          value={searchBarText}
          onChange={(e) => onHandleChange(e.target.value)}
          onKeyDown={
            (e) => {
              if ( e.key == 'Enter'){
              setIsValid(buttonAction(searchBarText))
            }
            if ( e.key == 'Backspace'){
             onHandleChange(""); setIsValid(undefined)
            }
            }          
          }
        />
        <button
          className="searchButton"
          onClick={() => {
            setIsValid(buttonAction(searchBarText));
          }}
        >
          <SearchIcon
            fill="#121212"
            stroke="white"
            width="18.51"
            height="18.51"
          />
        </button>
      </div>
      <div className="textFaded">{footerText}</div>
    </>
  );
}

export default SearchBar;
