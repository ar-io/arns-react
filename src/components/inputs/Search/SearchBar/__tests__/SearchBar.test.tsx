import { act, cleanup, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Arweave from 'arweave';
import { HashRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

import { PDNSContractCache } from '../../../../../services/arweave/PDNSContractCache';
import { WarpDataProvider } from '../../../../../services/arweave/WarpDataProvider';
import { ArweaveTransactionID, TRANSACTION_TYPES } from '../../../../../types';
import { DEFAULT_PDNT_CONTRACT_STATE } from '../../../../../utils/constants';
import { SearchBarFooter, SearchBarHeader } from '../../../../layout';
import {
  searchBarSuccessPredicate,
  searchBarValidationPredicate,
} from '../../../../pages/Home/Home';
import SearchBar from '../SearchBar';

Object.defineProperty(global.performance, 'markResourceTiming', {
  value: jest.fn(),
  writable: true,
});

const TEST_RECORDS = {
  ardrive: {
    contractTxId: 'I-cxQhfh0Zb9UqQNizC9PiLC41KpUeA9hjiVV02rQRw',
    endTimestamp: 1711122739,
    tier: 'SEC0-8cTfyDBRQo21KNIhUV5KreuEmIY05wX-VOeESE',
    type: TRANSACTION_TYPES.BUY,
  },
  'xn--go8h6v': {
    contractTxId: 'I-cxQhfh0Zb9UqQNizC9PiLC41KpUeA9hjiVV02rQRw',
    endTimestamp: 1711122739,
    tier: 'SEC0-8cTfyDBRQo21KNIhUV5KreuEmIY05wX-VOeESE',
    type: TRANSACTION_TYPES.LEASE,
  },
};

jest.mock('../../../../../hooks', () => ({
  useAuctionInfo: jest.fn(() => ({})),
  useIsFocused: jest.fn(() => ({})),
  useIsMobile: jest.fn(() => false),
  useArweaveCompositeProvider: jest.fn(),
  useWalletAddress: jest.fn(() => ({
    walletAddress: undefined,
    wallet: undefined,
  })),
  useRegistrationStatus: jest.fn(() => ({
    isDomainAvailable: false,
    isDomainInvalid: false,
    isDomainReserved: false,
  })),
}));

jest.mock('../../../../../state/contexts/GlobalState', () => ({
  useGlobalState: jest.fn(() => [
    {
      gateway: 'https://arweave.net',
      blockHeight: 1711122739,
      pdnsSourceContract: {},
    },
  ]),
}));

describe('SearchBar', () => {
  afterEach(cleanup);

  //const cache = new PDNSContractCache('some_url');

  // Spy on the methods and provide mock implementations
  // jest
  //   .spyOn(cache, 'getContractState')
  //   // eslint-disable-next-line
  //   .mockImplementation(async (id: ArweaveTransactionID) => {
  //     return DEFAULT_PDNT_CONTRACT_STATE;
  //   });

  // const arweave = Arweave.init({});

  // const warp = new WarpDataProvider(arweave);

  // // Spy on the methods and provide mock implementations
  // jest
  //   .spyOn(warp, 'getContractState')
  //   // eslint-disable-next-line
  //   .mockImplementation(async (id: ArweaveTransactionID) => {
  //     return DEFAULT_PDNT_CONTRACT_STATE;
  //   });

  const onChange = jest.fn();
  const onSubmit = jest.fn();
  const onFailure = jest.fn();
  const onSuccess = jest.fn();

  const searchBar = (
    <Router>
      <SearchBar
        onSubmit={onSubmit}
        onChange={onChange}
        onFailure={onFailure}
        onSuccess={onSuccess}
        successPredicate={(value: string | undefined) =>
          searchBarSuccessPredicate({ value, records: TEST_RECORDS })
        }
        validationPredicate={(value: string | undefined) =>
          searchBarValidationPredicate({ value })
        }
        value=""
        values={TEST_RECORDS}
        placeholderText={'Find a name'}
        headerElement={
          <SearchBarHeader defaultText="Find a name" reservedList={[]} />
        }
        footerElement={<SearchBarFooter isAuction={false} reservedList={[]} />}
      />
    </Router>
  );

  const { getByTestId } = render(searchBar);
  const searchInput = getByTestId('searchbar-input-id');
  const searchButton = getByTestId('search-button'); // Assuming you used data-testid="search-button"

  test('handles a capitalized name correctly', async () => {
    const domain = 'ARDRIVE';

    act(async () => {
      await userEvent.type(searchInput, domain);
      expect(onChange).toHaveBeenCalledWith();
      await userEvent.click(searchButton);

      expect(onSubmit).toHaveBeenCalled();
      expect(onFailure).toHaveBeenCalled();
    });
    const search = renderer
      .create(
        <Router>
          <SearchBar
            onSubmit={onSubmit}
            onChange={onChange}
            onFailure={onFailure}
            onSuccess={onSuccess}
            successPredicate={(value: string | undefined) =>
              searchBarSuccessPredicate({ value, records: TEST_RECORDS })
            }
            validationPredicate={(value: string | undefined) =>
              searchBarValidationPredicate({ value })
            }
            value={domain}
            values={TEST_RECORDS}
            placeholderText={'Find a name'}
            headerElement={
              <SearchBarHeader defaultText="Find a name" reservedList={[]} />
            }
            footerElement={
              <SearchBarFooter isAuction={false} reservedList={[]} />
            }
          />
        </Router>,
      )
      .toJSON();

    expect(search).toMatchSnapshot();
  });

  test('handles a lowercase name correctly', async () => {
    const domain = 'ardrive';

    act(async () => {
      await userEvent.type(searchInput, domain);
      await userEvent.click(searchButton);
      expect(onSubmit).toHaveBeenCalled();
      expect(onFailure).toHaveBeenCalled();
    });

    const search = renderer
      .create(
        <Router>
          <SearchBar
            onSubmit={onSubmit}
            onChange={onChange}
            onFailure={onFailure}
            onSuccess={onSuccess}
            successPredicate={(value: string | undefined) =>
              searchBarSuccessPredicate({ value, records: TEST_RECORDS })
            }
            validationPredicate={(value: string | undefined) =>
              searchBarValidationPredicate({ value })
            }
            value={domain}
            values={TEST_RECORDS}
            placeholderText={'Find a name'}
            headerElement={
              <SearchBarHeader defaultText="Find a name" reservedList={[]} />
            }
            footerElement={
              <SearchBarFooter isAuction={false} reservedList={[]} />
            }
          />
        </Router>,
      )
      .toJSON();

    expect(search).toMatchSnapshot();
  });
});
