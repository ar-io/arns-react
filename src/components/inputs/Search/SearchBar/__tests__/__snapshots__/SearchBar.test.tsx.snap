// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar handles a capitalized name correctly 1`] = `
<DocumentFragment>
  <div
    class="searchbar-container flex-center"
    style="max-width: 787px;"
  >
    <div
      class="searchbar"
      style="margin-bottom: 30px; width: 100%; position: relative;"
    >
      <div
        class=""
        id="validation-input"
        style="width: 100%;"
      >
        <div
          class="flex"
          style="width: 100%; position: relative;"
        >
          <input
            class="searchbar-input"
            data-testid="searchbar-input-id"
            enterkeyhint="search"
            id="searchbar-input-id"
            pattern="^[a-zA-Z0-9-]{0,32}$"
            placeholder="Find a name"
            spellcheck="false"
            type="search"
            value="ardrive"
          />
          <div
            style="position: absolute; right: 10px; top: 25%;"
          >
            <span />
          </div>
        </div>
        <div
          class="flex flex-column"
          style="font-size: 12px; display: flex; flex-direction: row; justify-content: space-between; margin-top: 15px; gap: 15px; z-index: 1001;"
        >
          <span
            class="flex flex-column flex-left text white"
            style="gap: 0px; align-items: flex-start; justify-content: flex-start;"
          >
            <span
              class="flex"
              style="gap: 12px; align-items: center;"
            >
              <span
                aria-label="check-circle"
                class="anticon anticon-check-circle"
                role="img"
                style="font-size: 16px;"
              >
                <svg
                  aria-hidden="true"
                  data-icon="check-circle"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm193.5 301.7l-210.6 292a31.8 31.8 0 01-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.2 0 19.9 4.9 25.9 13.3l71.2 98.8 157.2-218c6-8.3 15.6-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.5 12.7z"
                  />
                </svg>
              </span>
              Min. 1 character
            </span>
          </span>
          <span
            class="flex flex-column flex-left text white"
            style="gap: 0px; align-items: flex-start; justify-content: flex-start;"
          >
            <span
              class="flex"
              style="gap: 12px; align-items: center;"
            >
              <span
                aria-label="check-circle"
                class="anticon anticon-check-circle"
                role="img"
                style="font-size: 16px;"
              >
                <svg
                  aria-hidden="true"
                  data-icon="check-circle"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm193.5 301.7l-210.6 292a31.8 31.8 0 01-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.2 0 19.9 4.9 25.9 13.3l71.2 98.8 157.2-218c6-8.3 15.6-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.5 12.7z"
                  />
                </svg>
              </span>
              Max. 32 characters
            </span>
          </span>
          <span
            class="flex flex-column flex-left text white"
            style="gap: 0px; align-items: flex-start; justify-content: flex-start;"
          >
            <span
              class="flex"
              style="gap: 12px; align-items: center;"
            >
              <span
                aria-label="check-circle"
                class="anticon anticon-check-circle"
                role="img"
                style="font-size: 16px;"
              >
                <svg
                  aria-hidden="true"
                  data-icon="check-circle"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm193.5 301.7l-210.6 292a31.8 31.8 0 01-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.2 0 19.9 4.9 25.9 13.3l71.2 98.8 157.2-218c6-8.3 15.6-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.5 12.7z"
                  />
                </svg>
              </span>
              No special characters
            </span>
          </span>
          <span
            class="flex flex-column flex-left text white"
            style="gap: 0px; align-items: flex-start; justify-content: flex-start;"
          >
            <span
              class="flex"
              style="gap: 12px; align-items: center;"
            >
              <span
                aria-label="check-circle"
                class="anticon anticon-check-circle"
                role="img"
                style="font-size: 16px;"
              >
                <svg
                  aria-hidden="true"
                  data-icon="check-circle"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm193.5 301.7l-210.6 292a31.8 31.8 0 01-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.2 0 19.9 4.9 25.9 13.3l71.2 98.8 157.2-218c6-8.3 15.6-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.5 12.7z"
                  />
                </svg>
              </span>
              Dashes cannot be leading or trailing
            </span>
          </span>
        </div>
      </div>
      <button
        class="link bold text flex-center"
        style="position: absolute; right: 75px; height: 100%;"
      >
        Clear
      </button>
      <button
        class="button pointer"
        data-testid="search-button"
        style="border-left: 1px solid #38393B;"
      >
        <div
          fill="white"
          height="18"
          width="18"
        />
      </button>
    </div>
  </div>
  ,
</DocumentFragment>
`;

exports[`SearchBar handles a lowercase name correctly 1`] = `
<DocumentFragment>
  <div
    class="searchbar-container flex-center"
    style="max-width: 787px;"
  >
    <div
      class="searchbar"
      style="margin-bottom: 30px; width: 100%; position: relative;"
    >
      <div
        class=""
        id="validation-input"
        style="width: 100%;"
      >
        <div
          class="flex"
          style="width: 100%; position: relative;"
        >
          <input
            class="searchbar-input"
            data-testid="searchbar-input-id"
            enterkeyhint="search"
            id="searchbar-input-id"
            pattern="^[a-zA-Z0-9-]{0,32}$"
            placeholder="Find a name"
            spellcheck="false"
            type="search"
            value="ardrive"
          />
          <div
            style="position: absolute; right: 10px; top: 25%;"
          >
            <span />
          </div>
        </div>
        <div
          class="flex flex-column"
          style="font-size: 12px; display: flex; flex-direction: row; justify-content: space-between; margin-top: 15px; gap: 15px; z-index: 1001;"
        >
          <span
            class="flex flex-column flex-left text white"
            style="gap: 0px; align-items: flex-start; justify-content: flex-start;"
          >
            <span
              class="flex"
              style="gap: 12px; align-items: center;"
            >
              <span
                aria-label="check-circle"
                class="anticon anticon-check-circle"
                role="img"
                style="font-size: 16px;"
              >
                <svg
                  aria-hidden="true"
                  data-icon="check-circle"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm193.5 301.7l-210.6 292a31.8 31.8 0 01-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.2 0 19.9 4.9 25.9 13.3l71.2 98.8 157.2-218c6-8.3 15.6-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.5 12.7z"
                  />
                </svg>
              </span>
              Min. 1 character
            </span>
          </span>
          <span
            class="flex flex-column flex-left text white"
            style="gap: 0px; align-items: flex-start; justify-content: flex-start;"
          >
            <span
              class="flex"
              style="gap: 12px; align-items: center;"
            >
              <span
                aria-label="check-circle"
                class="anticon anticon-check-circle"
                role="img"
                style="font-size: 16px;"
              >
                <svg
                  aria-hidden="true"
                  data-icon="check-circle"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm193.5 301.7l-210.6 292a31.8 31.8 0 01-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.2 0 19.9 4.9 25.9 13.3l71.2 98.8 157.2-218c6-8.3 15.6-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.5 12.7z"
                  />
                </svg>
              </span>
              Max. 32 characters
            </span>
          </span>
          <span
            class="flex flex-column flex-left text white"
            style="gap: 0px; align-items: flex-start; justify-content: flex-start;"
          >
            <span
              class="flex"
              style="gap: 12px; align-items: center;"
            >
              <span
                aria-label="check-circle"
                class="anticon anticon-check-circle"
                role="img"
                style="font-size: 16px;"
              >
                <svg
                  aria-hidden="true"
                  data-icon="check-circle"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm193.5 301.7l-210.6 292a31.8 31.8 0 01-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.2 0 19.9 4.9 25.9 13.3l71.2 98.8 157.2-218c6-8.3 15.6-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.5 12.7z"
                  />
                </svg>
              </span>
              No special characters
            </span>
          </span>
          <span
            class="flex flex-column flex-left text white"
            style="gap: 0px; align-items: flex-start; justify-content: flex-start;"
          >
            <span
              class="flex"
              style="gap: 12px; align-items: center;"
            >
              <span
                aria-label="check-circle"
                class="anticon anticon-check-circle"
                role="img"
                style="font-size: 16px;"
              >
                <svg
                  aria-hidden="true"
                  data-icon="check-circle"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm193.5 301.7l-210.6 292a31.8 31.8 0 01-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.2 0 19.9 4.9 25.9 13.3l71.2 98.8 157.2-218c6-8.3 15.6-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.5 12.7z"
                  />
                </svg>
              </span>
              Dashes cannot be leading or trailing
            </span>
          </span>
        </div>
      </div>
      <button
        class="link bold text flex-center"
        style="position: absolute; right: 75px; height: 100%;"
      >
        Clear
      </button>
      <button
        class="button pointer"
        data-testid="search-button"
        style="border-left: 1px solid #38393B;"
      >
        <div
          fill="white"
          height="18"
          width="18"
        />
      </button>
    </div>
  </div>
  ,
</DocumentFragment>
`;

exports[`SearchBar renders correctly 1`] = `
<DocumentFragment>
  <div
    class="searchbar-container flex-center"
    style="max-width: 787px;"
  >
    <div
      class="searchbar"
      style="margin-bottom: 30px; width: 100%; position: relative;"
    >
      <div
        class=""
        id="validation-input"
        style="width: 100%;"
      >
        <div
          class="flex"
          style="width: 100%; position: relative;"
        >
          <input
            class="searchbar-input"
            data-testid="searchbar-input-id"
            enterkeyhint="search"
            id="searchbar-input-id"
            pattern="^[a-zA-Z0-9-]{0,32}$"
            placeholder="Find a name"
            spellcheck="false"
            type="search"
            value=""
          />
          <div
            style="position: absolute; right: 10px; top: 25%;"
          >
            <span />
          </div>
        </div>
        <div
          class="flex flex-column"
          style="font-size: 12px; display: flex; flex-direction: row; justify-content: space-between; margin-top: 15px; gap: 15px; z-index: 1001;"
        />
      </div>
      <button
        class="button pointer"
        data-testid="search-button"
        style="border-left: 1px solid #38393B;"
      >
        <div
          fill="white"
          height="18"
          width="18"
        />
      </button>
    </div>
  </div>
  ,
</DocumentFragment>
`;
